contracts/examples/dao/DAO.sol

```diff
@@ -119,47 +118,44 @@ contract DAO is VentureEth {
     /** Dividend distribution */
 
     /**
-     * @notice Propose to release DAO dividends.
+     * @notice Hook for proposals to release dividends.
      * @param amount The ether amount to be released as dividends.
      */
-    function proposeDividends(uint256 amount) public {
-        require(currentState == "LIVE", "DAO needs to be LIVE");
-        Voting voting = new Voting(
-            address(this),
-            address(this),
-            abi.encodeWithSignature("releaseDividends(uint256)", amount),
-            threshold
-        );
-        proposals.add(address(voting));
-        emit DividendsProposed(address(voting));
+    function releaseDividends(uint256 amount) public onlyProposal {
+        _releaseDividends(amount);
     }
 
+    /** Restart investor round */
+
     /**
-     * @notice Hook for proposals to release dividends.
-     * @param amount The ether amount to be released as dividends.
+     * @notice Hook for proposals to restart investor rounds.
      */
-    function releaseDividends(uint256 amount) public {
-        require(
-            proposals.contains(msg.sender),
-            "Only a proposal can execute."
-        );
-        proposals.remove(msg.sender);
-        _releaseDividends(amount);
+    function restartInvestorRound(uint256 _issuePrice) public onlyProposal {
+        _transition("SETUP");
+        this.setIssuePrice(_issuePrice);
+        this.startIssuance();
     }
 
-    /** Enumerators */
+    /**
+     * @notice Hook for proposals to start distribution in a non-initial investment round.
+     */
+    function restartDistribution() public onlyProposal {
```

Comment:
Maybe we could do an `if` in `startDistribution` instead.
`if (owner() != address(this)) transferOwnership(address(this))`

Reply:
startDistribution can be initially called only by the owner, while restartDistribution only by a voting.

Reply:
Got it, thanks.